ui/src/components/retreat/home/PraiseWall.js

```diff
@@ -0,0 +1,280 @@
+import {
+  Button,
+  Container,
+  Flex,
+  Heading,
+  Image,
+  Spacer,
+  SimpleGrid,
+  Stack,
+  Text,
+  Box,
+  VStack,
+  Icon,
+  Modal,
+  ModalOverlay,
+  Textarea,
+  useDisclosure,
+  ModalContent,
+  ModalHeader,
+  Alert,
+  AlertIcon,
+  AlertTitle,
+  AlertDescription,
+  FormControl,
+  FormLabel,
+  Input,
+  FormHelperText,
+  FormErrorMessage,
+  HStack,
+  ModalFooter,
+  CloseButton,
+  Wrap,
+  WrapItem,
+  ModalBody,
+  ModalCloseButton,
+  Center,
+} from '@chakra-ui/react';
+import { AddIcon } from '@chakra-ui/icons';
+import { BiNote } from 'react-icons/bi';
+
+import React, { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { customAxios as axios } from '../../helpers/customAxios';
+import PraiseCard from './PraiseCard';
+
+const AlertMessage = () => {
+  const [show, setShow] = useState(true);
+  useEffect(() => {
+    const timeId = setTimeout(() => {
+      setShow(false);
+    }, 5000);
+    return () => {
+      clearTimeout(timeId);
+    };
+  }, []);
+  if (!show) {
+    return null;
+  }
+  return (
+    <Alert status="success">
+      <AlertIcon />
+      <AlertDescription>Praise submitted successfully!</AlertDescription>
+    </Alert>
+  );
+};
+
+const PraiseWall = (props) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [input, setInput] = useState('');
+  const { user } = props;
+  const handleInputChange = (e) => setInput(e.target.value);
+
+  const [praiseList, setPraiseList] = useState([]);
+  const [formSubmitted, setFormSubmitted] = useState(false);
+  const { handleSubmit, register } = useForm();
+
+  const cardBgColors = ['#6dced3', '#ffdc82', '#f39371', '#ffdc82', '#6dced3'];
+  const eventCategory = 'CWC 2022';
+
+  const SplitToChunks = (array, parts) => {
+    let result = [];
+    for (let i = parts; i > 0; i--) {
+      result.push(array.splice(0, Math.ceil(array.length / i)));
+    }
+    return result;
+  };
+
+  const GetPraise = async () => {
+    try {
+      const { data, status } = await axios.get('/api/praises/get');
+
+      if (status === 200) {
+        data.sort((a, b) => (a.createdAt < b.createdAt ? 1 : -1));
+        setPraiseList([...data]);
+      } else {
+        throw Error('Something went wrong');
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const PostPraise = async ({ message }) => {
+    try {
+      const { data } = await axios.post('/api/praises/create', {
+        fullName: user.fullName,
+        message: message,
+        category: eventCategory,
+      });
+      setFormSubmitted(true);
```

Comment:
is this state ever switched back to false? wouldn't this make it so people can't really resubmit

Reply:
added changestate to false 5 seconds after submission